docs/jenkins.md

```diff
@@ -0,0 +1,199 @@
+
+* ensure plugin tests (if any) run when `rake jenkins:unit` is called, see "the example plugin":https://github.com/theforeman/foreman_plugin_template/ and "testing a plugin":https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Testing for help
+* create a foreman_example.yaml file in foreman-infra/JJB
+  * https://github.com/theforeman/foreman-infra/tree/master/puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins
+  * This will create a "test_plugin_foreman_example_master" job in Jenkins to test the master branch.
+* ensure the job is green by fixing bugs, installing dependencies etc.
+* add hook to GitHub repo, see "GitHub repo hook":https://projects.theforeman.org/projects/foreman/wiki/Jenkins#Testing-develop
+
+An org admin must then:
+
+* add the repo to the "Bots team":https://github.com/orgs/theforeman/teams/bots/repositories with +write+ access
+
+### Smart proxy plugin testing
+
+Proxy plugins are tested like ordinary gems with tests run entirely from the plugin directory, installing the smart proxy as a dependency (via bundler's git support).  The "test_proxy_plugin_matrix job":https://ci.theforeman.org/job/test_proxy_plugin_matrix/ is usually used to test plugins in a generic way.
+
+Each plugin should have a job defined in JJB that calls test_proxy_plugin_matrix here: https://ci.theforeman.org/view/Plugins/
+
+#### Smart proxy plugin PR testing
+
+To test pull requests, a separate job is used that also takes the PR details: https://ci.theforeman.org/view/Plugins/job/test_proxy_plugin_pull_request/
+
+To set this up for a plugin, make the template + manifest addition as noted in the PR section below to configure the PR scanner.  Also enable the prprocessor webhook on the GitHub repo for immediate builds.
+
+### Adding a new smart proxy plugin
+
+For a plugin "smart_proxy_example", first create a job that tests the main (master or develop) branch.
+
+* ensure plugin tests run when doing `bundle install` and `rake test`, see "testing a plugin":https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Smart-Proxy_Plugin#Testing for help
+#* if different branches rely on different versions of smart proxy, specify `:branch` in Gemfile on those branches
+* create a smart_proxy_example.yaml file in foreman-infra/JJB
+#* https://github.com/theforeman/foreman-infra/tree/master/puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins
+#* This will create a "test_proxy_plugin_smart_proxy_example_master" job in Jenkins to test the master branch.
+* ensure the job is green by fixing bugs, installing dependencies etc.
+* add hook to GitHub repo, see "GitHub repo hook":https://projects.theforeman.org/projects/foreman/wiki/Jenkins#Testing-develop
+
+Next, set up PR testing for the plugin:
+
+* add template to foreman-infra/puppet/modules/slave/templates, update branch, job names etc.
```

Comment:
nope

Reply:
Not sure why this was marked as resolved because this is no longer relevant and can be dropped.